pkgs/sketch_pad/lib/docs.dart

```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:dartpad_shared/model.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_markdown/flutter_markdown.dart';
+import 'package:url_launcher/url_launcher.dart' as url_launcher;
+
+import 'model.dart';
+import 'theme.dart';
+import 'widgets.dart';
+
+class DocsWidget extends StatefulWidget {
+  final AppModel appModel;
+
+  const DocsWidget({
+    required this.appModel,
+    super.key,
+  });
+
+  @override
+  State<DocsWidget> createState() => _DocsWidgetState();
+}
+
+class _DocsWidgetState extends State<DocsWidget> {
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Container(
+      decoration: BoxDecoration(
+        color: theme.scaffoldBackgroundColor,
+        border: Border(
+          top: Divider.createBorderSide(
+            context,
+            width: 8.0,
+            color: theme.colorScheme.surface,
+          ),
+        ),
+      ),
+      padding: const EdgeInsets.all(denseSpacing),
+      child: Stack(
+        children: [
+          ValueListenableBuilder(
+            valueListenable: widget.appModel.currentDocs,
+            builder: (context, DocumentResponse? docs, _) {
+              // TODO: Consider showing propagatedType if not null.
+
+              var title = _cleanUpTitle(docs?.elementDescription);
+              if (docs?.deprecated == true) {
+                title = '$title (deprecated)';
+              }
+
+              // TODO: should the markdown text be selectable?
```

Comment:
Hmm. I think I've never wanted to copy text from API docs before, but perhaps from code snippets in the API docs? We can revisit it :)

Reply:
I've removed the todo; we can re-adjust based on any user feedback here.